### PR TITLE
V1.6

### PR DESCRIPTION
**Pull request by naveenr-egov**

### CODE REVIEW

**Review by naveen-egov — APPROVED** [review]

File: apps/health_campaign_field_worker_app/android/settings.gradle [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
-def flutterSdkPath = properties.getProperty("flutter.sdk")
-assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
-apply from: "$flutterSdkPath/packages/flutter_tools/gradle/app_plugin_loader.gradle"
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version "7.3.0" apply false
+    id "org.jetbrains.kotlin.android" version "1.8.22" apply false
+    id "com.google.gms.google-services" version "4.4.0" apply false
+    id "com.google.firebase.crashlytics" version "2.9.9" apply false
+}
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/android/settings.gradle:25]: Will upgrade in next release

File: apps/health_campaign_field_worker_app/lib/blocs/project/project.dart [hunk 2]
```diff
@@ -594,6 +605,25 @@ class ProjectBloc extends Bloc<ProjectEvent, ProjectState> {
...
+      int configuredBatchSize = getBatchSizeToBandwidth(
+        speed,
+        configs,
+        isDownSync: true,
+      );
+      return configuredBatchSize;
+    } catch (e) {
+      rethrow;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/lib/blocs/project/project.dart:626]: We have default bandwidth in env file

File: apps/health_campaign_field_worker_app/lib/pages/home.dart [hunk 3]
```diff
@@ -61,22 +66,18 @@ class HomePage extends LocalizedStatefulWidget {
 class _HomePageState extends LocalizedState<HomePage> {
   bool skipProgressBar = false;
   final storage = const FlutterSecureStorage();
-  late StreamSubscription<ConnectivityResult> subscription;
+  late StreamSubscription<List<ConnectivityResult>> subscription;
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/lib/pages/home.dart:69]: We already have it as list

File: apps/health_campaign_field_worker_app/lib/data/sync_service_mapper.dart [hunk 4]
```diff
@@ -0,0 +1,862 @@
...
+          }
+
+          return e.copyWith(taskId: serverGeneratedId);
+        }).toList(),
+      );
+    }
+
+    return updatedEntity;
+  }
+}
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/lib/data/sync_service_mapper.dart:862]: The current implementation works as intended

File: apps/health_campaign_field_worker_app/lib/utils/i18_key_constants.dart [hunk 5]
```diff
@@ -181,7 +180,7 @@ class HomeShowcase {
     return 'SUPERVISOR_HOME_SHOWCASE_PROGRESS_BAR';
   }
 
-  String get supervisorMyChecklist {
+  String get supervisorMySurveyForm {
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/lib/utils/i18_key_constants.dart:183]: It is already surveyForm please check. Don't give false comments.

File: apps/health_campaign_field_worker_app/lib/pages/boundary_selection.dart [hunk 6]
```diff
@@ -41,19 +43,21 @@ class _BoundarySelectionPageState
...
   void initState() {
+    context.syncRefresh();
     LocalizationParams().setModule('common', false);
     context.read<SyncBloc>().add(SyncRefreshEvent(context.loggedInUserUuid));
     context.read<BeneficiaryDownSyncBloc>().add(
           const DownSyncResetStateEvent(),
         );
     super.initState();
+    listenToSyncCount();
   }
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/lib/pages/boundary_selection.dart:61]: dispose will not be called before initstate in any usecase. This is not a valid suggestion.

File: apps/health_campaign_field_worker_app/lib/pages/boundary_selection.dart [hunk 7]
```diff
@@ -525,41 +536,31 @@
...
+                                                          .common
+                                                          .corecommonRequired),
+                                                      true,
+                                                      Theme.of(context),
+                                                    ),
+                                                  );
+                                                } else {
+                                                  setState(() {
+                                                    shouldPop = true;
+                                                  });
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/lib/pages/boundary_selection.dart:563]: It's just 2 conditions one is form validation and the other is role check and mounted. It's pretty clear. If it exceeds more will update it then.

File: apps/health_campaign_field_worker_app/lib/widgets/showcase/config/home_showcase.dart [hunk 8]
```diff
@@ -19,7 +19,7 @@ class _HomePageShowcaseData {
         warehouseManagerSyncData,
         inventoryReport,
         supervisorProgressBar,
-        supervisorMyChecklist,
+        supervisorMySurveyForm,
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/lib/widgets/showcase/config/home_showcase.dart:22]: It is already updated to surveyForm. Dont give false comments.

File: apps/health_campaign_field_worker_app/lib/data/repositories/remote/bandwidth_check.dart [hunk 9]
```diff
@@ -29,7 +29,7 @@ class BandwidthCheckRepository {
       final speed = (((800 / timeconsumed) / 1000));
 
       return speed;
-    } on DioError catch (e) {
+    } on DioException catch (e) {
       rethrow;
     }
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/lib/data/repositories/remote/bandwidth_check.dart:34]: This currently doesnt require domain specific ignoring for now.

File: apps/health_campaign_field_worker_app/android/app/src/main/kotlin/com/digit/hcm/MainActivity.kt [hunk 10]
```diff
@@ -1,60 +1,82 @@
...
             // Optionally, you can send this data to Flutter via MethodChannel
             flutterEngine?.dartExecutor?.binaryMessenger?.let {
-                MethodChannel(it, CHANNEL).invokeMethod("locationUpdate", mapOf("latitude" to latitude, "longitude" to longitude))
+                MethodChannel(it, CHANNEL).invokeMethod(
+                    "locationUpdate", mapOf(
+                        "latitude" to latitude,
+                        "longitude" to longitude,
+                        "accuracy" to accuracy
+                    )
+                )
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply apps/health_campaign_field_worker_app/android/app/src/main/kotlin/com/digit/hcm/MainActivity.kt:35]: Added null check

File: packages/attendance_management/lib/blocs/attendance_bloc.dart [hunk 11]
```diff
@@ -84,7 +83,7 @@ class AttendanceBloc extends Bloc<AttendanceEvents, AttendanceStates> {
     emit(const RegisterLoading());
     // Finding and emitting the selected register based on ID
     final selectedRegister =
-        await event.registers.where((e) => e.id == event.registerID).first;
+        event.registers.where((e) => e.id == event.registerID).first;
```
[the remark this answers is not in the document]
Comment (naveenr-egov) [reply packages/attendance_management/lib/blocs/attendance_bloc.dart:86]: Selecting the result only after displayed on the UI. This usecase never occurs.

**Pull request merged**